yb-voyager/src/utils/utils.go

```diff
@@ -388,16 +389,23 @@ func WaitForLineInLogFile(filePath string, message string, timeoutDuration time.
 	defer file.Close()
 
 	for {
-		scanner := bufio.NewScanner(file)
-		for scanner.Scan() {
-			line := scanner.Text()
-			if strings.Contains(line, message) {
-				return nil
+		reader := bufio.NewReader(file)
+		for {
+			line, err := reader.ReadString('\n')
```

Comment:
@makalaaneesh can you explain why that error would come with scanner but not reader.

Both of these packages having something like default buffer size.
I am not sure if this will also not break in case of large string.

Also refer `reader.ReadLine()` function once. That seems more full proof to me.
We also have `utils.Readline()` in our code which is not used anywhere anymore

Reply:
@sanyamsinghal from the docs of [Scanner](https://pkg.go.dev/bufio#Scanner): 
`
Scanning stops unrecoverably at EOF, the first I/O error, or a token too large to fit in the [Scanner.Buffer](https://pkg.go.dev/bufio#Scanner.Buffer). When a scan stops, the reader may have advanced arbitrarily far past the last token. Programs that need more control over error handling or large tokens, or must run sequential scans on a reader, should use [bufio.Reader](https://pkg.go.dev/bufio#Reader) instead.
` 
The error we were running into was because of large tokens, hence why I started exploring the `Reader`. 

wrt [Reader.Readline](https://pkg.go.dev/bufio#Reader.ReadLine)  : 
from docs: `ReadLine is a low-level line-reading primitive. Most callers should use Reader.ReadBytes('\n') or Reader.ReadString('\n') instead or use a [Scanner](https://pkg.go.dev/bufio#Scanner).`

Moreover, `If the line was too long for the buffer then isPrefix is set and the beginning of the line is returned. The rest of the line will be returned from future calls. isPrefix will be false when returning the last fragment of the line. ` which basically means we need extra handling to process a line. (which is what `utils.Readline` in our code is doing). 

It seems like like `Reader.ReadString` abstracts all this away, and serves our purpose at least.